nautilus_open_any_terminal/open_any_terminal_extension.py

```diff
@@ -183,28 +192,19 @@ def _checkdecode(s):
 
 def open_terminal_in_file(filename):
     """open the new terminal with correct path"""
-    if filename:
-        # escape filename quotations
-        filename = filename.replace('"', '\\"')
-        if new_tab:
-            call(
-                '{0} {1} {2}"{3}" &'.format(
-                    terminal_cmd,
-                    NEW_TAB_PARAMS[terminal],
-                    TERM_WORKDIR_PARAMS[terminal],
-                    filename,
-                ),
-                shell=True,
-            )
-        else:
-            call(
-                '{0} {1}"{2}" &'.format(
-                    terminal_cmd, TERM_WORKDIR_PARAMS[terminal], filename
-                ),
-                shell=True,
-            )
-    else:
-        call("{0} &".format(terminal_cmd), shell=True)
+    cmd = terminal_cmd.copy()
+    cmd[0] = shutil.which(cmd[0])
+    if new_tab:
+        cmd.append(NEW_TAB_PARAMS[terminal])
+
+    cwd_params = TERM_WORKDIR_PARAMS.get(terminal)
+    if cwd_params and filename:
+        cmd.extend(cwd_params)
+        if terminal == "blackbox":
+            # This is required
+            cmd.append(filename)
```

Comment:
can you please explain this part? not sure about the inner if block

Reply:
blackbox, unlike other terminals, overrides the current directory and cannot be overridden with a dot path. So as a workaround I pass the directory on the command line. Other terminals either keep cwd or understand dot paths, so I don't need to append anything for them.

Reply:
not your fault, but I am not happy with `blackbox` in this plugin at all. it adds complexity where all other programs behave normally: #99, #61

But sure lets leave it like this.